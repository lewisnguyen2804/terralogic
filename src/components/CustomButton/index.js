import React, { Component } from 'react'

export default class CustomButton extends Component {
    render() {
        const { 
            type = 'button', 
            className = '', 
            onClick = {}, 
            value = "Button" 
        } = this.props;
        return (
            <input
                type={type}
                className={className}
                onClick={onClick}
                value={value} />
        )
    }
}
