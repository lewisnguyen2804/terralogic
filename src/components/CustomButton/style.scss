@import '../../style.scss';


// BUTTONS
.main-button {
    border: none;
    border-radius: 30px;
    height: 60px;
    @extend .bold-font;
    font-size: 1.2rem;
    outline: none;
}

.button-type-1 {
    @extend .main-button;
    background-color: $third-color;
    color: white;
    box-shadow: 0 10px 20px 0 rgba(189, 16, 224, 0.3);
    &:hover {
        background-color: $main-color;
        transition: 0.3s;
    }
}

.button-type-2 {
    @extend .main-button;
    background-color: white;
    color: $third-color;
    border: 2px solid $third-color;
    &:hover {
        background-color: $main-color;
        border: 2px solid $main-color;
        color: white;
        transition: 0.3s;
    }
}

.space {
    width: 20px
}