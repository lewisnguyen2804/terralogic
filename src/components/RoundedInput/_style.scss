
.input-container {
    margin-top: 20px;
}

.input-title {
    margin-left: 12px;
    font-weight: bold;
    color: $gray-color;
}
.input-with-icon {
    position: relative;
    .input-icon {
        position: absolute;
        left: 12px;
        top: 12px;
    }
    .input-text {
        text-indent: 45px;
        outline: none;
        width: 100%;
        height: 40px;
        border-radius: 20px;
        border: 1px solid #dfe2e6;
        &::placeholder {
            color: #dfe2e6;
        }
        &:focus {
            border: 1px solid $third-color;
        }
    }
    .show-password-icon {
        position: absolute;
        right: 12px;
        top: 13px;
        cursor: pointer;
     }
    .hide-password-icon {
        @extend .show-password-icon;
        // change color
    }
}