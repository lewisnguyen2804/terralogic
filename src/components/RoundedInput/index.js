import React, { Component } from 'react'
import './style.scss'

export default class RoundedInput extends Component {
    render() {
        return (
            <div className="input-container">
                <p className="input-title">{this.props.title}</p>
                <div className="input-with-icon">
                    <img className="input-icon" alt="input-icon" src={this.props.icon} />
                    <input
                        className="input-text"
                        type={this.props.type}
                        value={this.props.value}
                        placeholder={this.props.placeholder}
                        name={this.props.name}
                        onChange={this.props.handleChange}
                    />
                    {this.props.isPassword === 'true' ? <img className={this.props.showPwdClassName} alt="pwd-icon" onClick={this.props.showPwdOnClick} src={this.props.showPwdIcon} />:''}
                </div>
            </div>
        )
    }
}
