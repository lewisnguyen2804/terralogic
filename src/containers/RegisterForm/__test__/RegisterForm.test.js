import React from 'react'
import { shallow } from 'enzyme';
import { RegisterForm } from '../';
import RoundedInput from '../../../components/RoundedInput';
import Enzyme from 'enzyme';
import CustomButton from '../../../components/CustomButton';

describe('test for Register Page', () => {
    const wrapper = shallow(<RegisterForm />);
    it('should render correctly', () => {
        expect(JSON.stringify(wrapper)).toMatchSnapshot();
    });
});

// handleSubmit function
describe('test for Register Page', () => {
    const wrapper = shallow(<RegisterForm />);
    const instance = wrapper.instance();
    const registerClick = jest.spyOn(instance, 'handleSubmit');
    wrapper.update();
    instance.forceUpdate();
    it('it should render with check func', () => {
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(registerClick).toHaveBeenCalled();
    });
});


// handleChange function
describe('test for Register Page', () => {
    const wrapper = shallow(<RegisterForm />);
    const instance = wrapper.instance();
    const handleChangeClick = jest.spyOn(instance, 'handleChange');
    wrapper.update();
    instance.forceUpdate();
    it('it should render with check func', () => {
        const inputTest = Enzyme.mount(<RoundedInput handleChange={handleChangeClick} value="custom value" />);
        inputTest.find('input').simulate('change');
        expect(handleChangeClick).toHaveBeenCalled();
    });
});


// onShowPasswordClick/onShowConfirmPasswordClick function
describe('test for RegisterForm Page', () => {
    const wrapper = shallow(<RegisterForm />);
    const instance = wrapper.instance();
    const onShowPasswordClick = jest.spyOn(instance, 'onShowPasswordClick');
    const onShowConfirmPasswordClick = jest.spyOn(instance, 'onShowConfirmPasswordClick');
    wrapper.update();
    instance.forceUpdate();
    it('it should render with check func', () => {
        const clickTest = Enzyme.mount(<RoundedInput isPassword="true" showPwdOnClick={onShowPasswordClick} />);
        clickTest.find('img').at(1).simulate('click');
        expect(onShowPasswordClick).toHaveBeenCalled();

        const clickTest1 = Enzyme.mount(<RoundedInput isPassword="true" showPwdOnClick={onShowConfirmPasswordClick} />);
        clickTest1.find('img').at(1).simulate('click');
        expect(onShowConfirmPasswordClick).toHaveBeenCalled();
    });
});


// goBack function
const props = {
    history:
    {
        goBack: jest.fn()
    }
}

describe('test for Register Page', () => {
    const wrapper = shallow(<RegisterForm {...props} />);
    const instance = wrapper.instance();
    const goBackClick = jest.spyOn(instance, 'goBack');
    wrapper.update();
    instance.forceUpdate();
    it('it should render with check func', () => {
        const inputTest = Enzyme.mount(<CustomButton onClick={goBackClick} />);
        expect(inputTest).toMatchSnapshot();
        inputTest.find('input').simulate('click');
        expect(inputTest).toHaveBeenCalled();
    });
});