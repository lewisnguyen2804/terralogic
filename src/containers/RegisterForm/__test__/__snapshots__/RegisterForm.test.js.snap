// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for Register Page it should render with check func 1`] = `ReactWrapper {}`;

exports[`test for Register Page should render correctly 1`] = `"{}"`;
